app/views/Project/Tagging/Export/NewExport/EntriesExportSelection/index.tsx

```diff
@@ -0,0 +1,481 @@
+
+    const [
+        reportStructureVariant,
+        setReportStructureVariant,
+    ] = useState<string>(SECTOR_FIRST);
+    const [
+        contextualWidgets,
+        setContextualWidgets,
+    ] = useState<TreeSelectableWidget<string>[]>([]);
+
+    const variables = useMemo(
+        (): ProjectFrameworkDetailsQueryVariables => ({
+            projectId,
+        }),
+        [projectId],
+    );
+
+    const {
+        loading: frameworkGetPending,
+        data: frameworkResponse,
+    } = useQuery<ProjectFrameworkDetailsQuery, ProjectFrameworkDetailsQueryVariables>(
+        PROJECT_FRAMEWORK_DETAILS,
+        {
+            variables,
+            onCompleted: (response) => {
+                // TODO handle for conditional widgets
+                const widgets = getWidgets(
+                    response.project?.analysisFramework as AnalysisFramework,
+                );
+                const textWidgetList = widgets
+                    ?.filter((v) => v.widgetId === 'TEXT')
+                    .map((v) => ({ ...v, selected: true }));
+
+                const contextualWidgetList = filterContexualWidgets(widgets)
+                    ?.map((v) => ({ ...v, selected: true }));
+
+                setTextWidgets(textWidgetList ?? []);
+                setContextualWidgets(contextualWidgetList ?? []);
+            },
+        },
+    );
+
+    const [
+        createExport,
+        {
+            loading: createExportPending,
+        },
+    ] = useMutation<CreateExportMutation, CreateExportMutationVariables>(
+        CREATE_EXPORT,
+        {
+            refetchQueries: [
+                'ProjectExports',
+            ],
+            onCompleted: (response) => {
+                if (response?.project?.exportCreate?.ok) {
+                    if (response.project.exportCreate.result?.isPreview) {
+                        setPreviewId(response.project.exportCreate.result?.id);
+                    } else {
+                        alert.show(
+                            _ts('export', 'exportStartedNotifyMessage'),
+                            { variant: 'success' },
+                        );
+                    }
+                }
+            },
+            onError: (gqlError) => {
+                alert.show(
+                    gqlError.message,
+                    {
+                        variant: 'error',
+                    },
+                );
+            },
+        },
+    );
+
+    const analysisFramework = frameworkResponse?.project?.analysisFramework as AnalysisFramework;
+    useEffect(() => {
+        const structure = createReportStructure(
+            reportStructureVariant,
+            includeSubSector,
+            analysisFramework,
+        );
+        setReportStructure(structure);
+    }, [
+        analysisFramework,
+        reportStructureVariant,
+        includeSubSector,
+    ]);
```

Comment:
Do we need useEffect here?

Reply:
we need to recreate the report structure when each of reportStructureVariant, includeSubSector and analysisFramework changes. there are separate functions to handle each of the above-mentioned variables. instead of calling createReportStructure in each handler function useEffect is simpler.